Zend/zend_compile.c

```diff
@@ -10589,6 +10589,25 @@ static void zend_compile_include_or_eval(znode *result, zend_ast *ast) /* {{{ */
 }
 /* }}} */
 
+static void zend_compile_void_cast(znode *result, zend_ast *ast)
+{
+	zend_ast *expr_ast = ast->child[0];
+	znode expr_node;
+	zend_op *opline;
+
+	zend_do_extended_fcall_begin();
+	zend_compile_expr(&expr_node, expr_ast);
+
+	switch (expr_node.op_type) {
+		case IS_TMP_VAR:
+		case IS_VAR:
+		case IS_CONST:
```

Comment:
Can't you just `zend_do_free(&expr_node);` when it's IS_CONST? Seems pointless to generate opcodes for that.
Then you can also avoid the VM change for ZEND_FREE (which wouldn't be optimal if it actually executed for CONSTs as you have the SAVE_OPLINE and exception check)

Reply:
> Can't you just `zend_do_free(&expr_node);` when it's IS_CONST?

Yes, I can.